### ISSUE TITLE
[BUG] Redis-Backed Chat Memory - Error: Connection is closed

### ISSUE THREAD

**Issue by ViniciusNoma** [FlowiseAI/Flowise#2047]
**Describe the bug**
I recently set up a workflow using OpenAI Function Agent with Redis-Backed Chat Memory. However, I noticed that sometimes 
when I save the project and go to another similar project also using Redis-Backed Chat Memory (with different connect credential) and try to interact with the chat, it generates the following error: Chain run errored with error: "Connection is closed.\n\nError: Connection is closed.\n    at EventEmitter.sendCommand".

It looks like the problem is with query param in the redis URL. Example: redis://localhost:6379?keyPrefix=something

**To Reproduce**

1. Create two projects with Redis-Backed Chat Memory
2. Setup local Redis
3. Interact with the chat

**Screenshots**
<img width="905" alt="Captura de Tela 2024-03-26 às 11 41 42" src="https://github.com/FlowiseAI/Flowise/assets/86965431/c895a170-4431-43dc-819c-e6d4252403a0">

**Setup**
-   OS: macOS
-   Browser:  chrome

**Comment by IgorBMattiolli** [comment]
vish

**Comment by ViniciusNoma** [comment]
@HenryHengZJ Yes, it is working outside of Flowise. I forgot to mention that I also tried connecting to local Redis without adding query parameters, and it worked.

I believe the problem lies with the memory node instance. I attempted to debug it, and when I save the project containing the Redis-Backed Chat Memory and switch to another project with the same node (both using the same connection), an error message appears: 'Connection is closed. Error: Connection is closed. at EventEmitter.sendCommand'.
